Msus test data

hsemwal opened the pull request:
Here, I will implement a test using real data from a different scope

Proof of Not-Working

I don't have this implemented so it doesn't really work.
plus the lint tests come back with failures all the time
Steps to complete

capture a small (<=5MB) sample of raw binary data coming out of the new scope, add it to the tests/data directory
make a test fixture that creates a version of your StreamDaq with default configuration that reads from that .bin file - [see the “default_streamdaq” fixture for an example](https://github.com/Aharoni-Lab/mio/blob/975151f102c2baaec25c1e9c436226b96757ad12/tests/test_stream_daq.py#L20) and [read the pytest docs re: fixtures](https://docs.pytest.org/en/latest/how-to/fixtures.html)
Current Status

I recorded .bin data
now going to try* and implement the code using some of the pre-existing code

<!-- readthedocs-preview miniscope-io start -->
----
📚 Documentation preview 📚: https://miniscope-io--115.org.readthedocs.build/en/115/

<!-- readthedocs-preview miniscope-io end -->

hsemwal committed:
Merge pull request #113 from Aharoni-Lab/MSUS_frame_process-jls

Processing gs frame data

hsemwal committed:
Added a test_stream to new scope

Added empty python file for testing .bin files

hsemwal committed:
Renamed and copied bin test function

Renamed the python file to test_stream_new_scope to avoid confusion with test_stream, and copied function test_bin_output to the python file.

hsemwal committed:
First Commit

Branching off properly for commit

hsemwal committed:
Setting up the imports

Removed conftest and finding a place for the .bin file

hsemwal committed:
Modifications to pytest

modifications to bin and storage

hsemwal committed:
adding .bin file to directory

Added some recorded .bin files to the mystery scope test directory

hsemwal committed:
.gitignore updated

updated gitignore to include .bin/.avi/.csv

sneakers-the-rat committed:
extract fpga_recv buffer splitting logic to separate iterator function, untested.

sneakers-the-rat committed:
Merge pull request #109 from Aharoni-Lab/bugfix-config-sources-property

Make `config_sources` a classmethod rather than a class property

sneakers-the-rat committed:
accessing the preserved buffer rather than the new buffer

sneakers-the-rat committed:
add tests

sneakers-the-rat committed:
python 3.9-3.12 compat

sneakers-the-rat committed:
incredible levels of python 3.12 compat

sneakers-the-rat committed:
Merge pull request #116 from Aharoni-Lab/streamdaq-iter-source

Separate buffer-splitting in `_fpga_recv` into a generator function

hsemwal committed:
Attempts at creating test with Raymond

Attempting to create .yml test file in the main test folder and call the configuration from the test_stream function

hsemwal committed:
GSStreamDaq adding configYamlmixin, config file correction

Implementing configyamlmixin file into class GSSStreamDaq, removed some repeated variables, and added descriptions/corrections for config.py for mystery scope properties

sneakers-the-rat commented:
Btw, so next step after #116 was merged is to then update these branches so you can use it. The MSUS branches are a little bit of chaos right now, like you are pulling into a branch of a branch of a branch here. Tidy up as you go so you don't grow a maze.

 so like I said in the prior PR:

- decide on a branch to use as the "main" branch for MSUS stuff - I recommend `msus`
- merge all the stuff you want into that branch, delete the other branches
- merge `main` into that branch.

Then as you work...
- make new branch.
- do a small amount of work in it
- merge that into the `msus` (or whatever you want to call it) branch

Until we've gotten it basically working. Only reason we aren't merging into `main` is educational reasons - giving you side branches to practice this flow.

sneakers-the-rat commented on tests/test_gs/test_stream_new_scope.py:
```diff
@@ -0,0 +1,39 @@
+
+    assert output_file.exists()
+
+    assert hash_file(data_file) == hash_file(output_file)
```
Haha they fuckin better match

hsemwal committed:
Timeout for debug

Previous timeout for debug set to 60 s

hsemwal committed:
Pixel Depth Final --> Config Matches Initial

Config should be set to the final values for the scope (320x320x10). Config should change to include all the data from the camera (328x320x12), so I included modification for increasing pixel value.

hsemwal committed:
Changed to mystery config

Changed daqconfig from streamDAQ to DevConfig

hsemwal committed:
Merge remote-tracking branch 'origin/main' into MSUS_test_data

hsemwal committed:
Test for stream_daq and header works

Removed the +1 (python is a 0 index program) from the header, allowing the tests for header and stream_daq for mystery scope to pass!

hsemwal committed:
Testing MysterScope iterator method

Testing the method for the mysteryscope

hsemwal committed:
Ok

not sure what any of this is doing but ok

hsemwal committed:
OK not sure what this is doing either

but whatever some higher ups want things to be done like this without explanation

hsemwal committed:
Notes to header

Reverted header (and tests) back to original, and added some comments to the header for future reference. At some point write_timestamps will need to be changed.

hsemwal committed:
Creating Fake Data Using Tests

Created fake data using tests, including patterns that came in from the size of the full and partial data buffers

hsemwal committed:
Changes to .yml files

The yml files are now set up to allow differences between generic scopes

hsemwal committed:
implementing changes with Jonny

implement binarydaq to collect binary samples before full conversion is working, set up binary sample test fixtures, fix indexing bug in header splitting

hsemwal committed:
implement binarydaq to collect binary samples before full conversion is working, set up binary sample test fixtures, fix indexing bug in header splitting

hsemwal committed:
Comments for todo

Added some general direction

hsemwal committed:
Comments for todo

Added some general direction

hsemwal committed:
No byteswap for header

Changed the byteswap for the header, resulting in correct header data

hsemwal committed:
Merge remote-tracking branch 'origin/MSUS_test_data' into MSUS_test_data

hsemwal committed:
Recordd 50 frame bright to dark bin file

Recorded a 50 frame bright to dark (hand covered, verified on oscilloscope) and updated gitignore with .bin file name (so we dont need to push)

hsemwal committed:
Recordd 50 frame bright to dark bin file

Recorded a 50 frame bright to dark (hand covered, verified on oscilloscope) and updated gitignore with .bin file name (so we dont need to push)

hsemwal committed:
Updated local gitignore and set to 15 nframes

Updated pdmbuild gitiginore and set up 15 nframes light to dark test

hsemwal committed:
Merge remote-tracking branch 'origin/MSUS_test_data' into MSUS_test_data

hsemwal committed:
Debugging size of headers

Debugging the sizes of headers in MIO, should be 12 (not 14)

hsemwal committed:
Increased speed and mio tests

Increased baud rate to 8Mhz and continuing to debug the header and data sizes

hsemwal committed:
Guess on header length

Guessing header length issues, and trying experiments on 15 frame sample.

hsemwal committed:
fix header length computation

hsemwal committed:
Fixed mystery header algo

Header algorithm wasn't correct, here fixed the size of the header length with @jonnys help

hsemwal committed:
Close to getting tests and fixing size calculation bugs

1. config files are not set to the final (processed) version of the camera output
2. test files have trouble constructing a frame
3.

Results: Here are the changes I (and Jonny) made in MSUS Mio today

1. Changed original config files
    1. MSUS and MSUS-test
2. Changed npix calculation
    1. daq.py—> buffer_npix
3. Added config modification files to ensure all data is inputted correctly
    1. buffer_npix calles self.config.frame_width_input
4. Changed
5. Corrected errors
    1. This includes buffer2array not returning data
6. Added test_daq

Todo:
Here are the things that I am working on:
1. fix the test to calculate the final processing is working

hsemwal committed:
Testing buffer_to_array function

Created a buffer_to_array testing function!

hsemwal committed:
MIO testing trim

Bigenning to investigate mio _trim method

sneakers-the-rat commented on pyproject.toml:
```diff
@@ -120,7 +120,7 @@ filterwarnings = [
     "tests"
 ]
-timeout = 60
+# timeout = 60
```
```suggestion
timeout = 60
```

I am uncommenting this because this PR keeps taking up all the ci runner slots because of breakpoints left in committed code and other non-termination problems.

sneakers-the-rat committed:
restore testing timeouts
